commit 297dc84acd5fcb1f981b171cff6e7b63624db0c1
Author: Johannes 'josch' Schauer <josch@mister-muffin.de>
Date:   Wed Oct 14 10:23:07 2015 +0200

    Fix regex parsing `schroot --info` output (closes: #801572)
    
     - this was a regression introduced by commit 8a3e12c3
     - this commit fixes the regression and also improves the regex

diff --git a/lib/Sbuild/ChrootInfoSchroot.pm b/lib/Sbuild/ChrootInfoSchroot.pm
index ac1bcc7..5bd6e0a 100644
--- a/lib/Sbuild/ChrootInfoSchroot.pm
+++ b/lib/Sbuild/ChrootInfoSchroot.pm
@@ -61,27 +61,15 @@ sub get_info_from_stream {
 
 	last if ! $_;
 
-	if (/\s*--- Chroot ---/ &&
+	if (/^\s*(───|---) Chroot \1$/ &&
 	    $tmp{'Namespace'} eq "") {
 	    $tmp{'Namespace'} = 'chroot';
 	}
-	if (/\s*--- Session ---/ &&
+	if (/^\s*(───|---) Session \1$/ &&
 	    $tmp{'Namespace'} eq "") {
 	    $tmp{'Namespace'} = 'session';
 	}
-	if (/\s*--- Source ---/ &&
-	    $tmp{'Namespace'} eq "") {
-	    $tmp{'Namespace'} = 'source';
-	}
-	if (/\s*--- Chroot ---/ &&
-	    $tmp{'Namespace'} eq "") {
-	    $tmp{'Namespace'} = 'chroot';
-	}
-	if (/\s*--- Session ---/ &&
-	    $tmp{'Namespace'} eq "") {
-	    $tmp{'Namespace'} = 'session';
-	}
-	if (/\s*--- Source ---/ &&
+	if (/^\s*(───|---) Source \1$/ &&
 	    $tmp{'Namespace'} eq "") {
 	    $tmp{'Namespace'} = 'source';
 	}
